src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -51,7 +51,30 @@ internal partial class Binder
                     // BindType for AttributeSyntax's name is handled specially during lookup, see Binder.LookupAttributeType.
                     // When looking up a name in attribute type context, we generate a diagnostic + error type if it is not an attribute type, i.e. named type deriving from System.Attribute.
                     // Hence we can assume here that BindType returns a NamedTypeSymbol.
-                    boundAttributeTypes[i] = (NamedTypeSymbol)binder.BindType(attributeToBind.Name, diagnostics).Type;
+                    var boundType = binder.BindType(attributeToBind.Name, diagnostics);
+                    var boundTypeSymbol = (NamedTypeSymbol)boundType.Type;
+
+                    // Check type arguments inside the attribute type (unless the attribute type is already an error).
+                    if (boundTypeSymbol.TypeKind != TypeKind.Error)
+                    {
+                        var location = attributeToBind.GetLocation();
+                        for (var type = boundTypeSymbol; type is not null; type = type.ContainingType)
```

Comment:
>for (var type = boundTypeSymbol; type is not null; type = type.ContainingType) [](http://example.com/codeflow?start=24&length=78)

I think all this logic should be moved into ```CheckDisallowedAttributeDependentType``` and it should be called with ```boundType```. ```CheckDisallowedAttributeDependentType``` already traverses containing types, we just need to add checks for unbound generic type and type parameters.

Reply:
Alright, but `CheckDisallowedAttributeDependentType` traverses the whole type, not just nested types, so after moving the checks there, I changed them to work directly on the type arguments and hence the reported argument slightly changed: 1480493cbd30fee714e29a50fbd6102d0c201a43